src/handle_action.cpp

```diff
@@ -386,12 +386,13 @@ input_context game::get_player_input( std::string &action )
                 .on_top( true );
             }
 
-#if defined(TILES)
             // Remove asynchronous animations after animation delay if no input
             if( current_turn.async_anim_timeout() ) {
+#if defined(TILES)
                 tilecontext->void_async_anim();
-            }
 #endif
+                g->void_async_anim_curses();
```

Comment:
Does it redraw in curses if you call `g->invalidate_main_ui_adaptor()` here instead of calling `ui_manager::redraw()` in `void_async_anim_curses`?